### PR TITLE
add canonical url and ID to spotlight indexing

### PR DESCRIPTION
**Pull request by ahmednawar**
@derrickstaten @aaustin @Sarkar

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BranchUniversalObject.m [hunk 1]
```diff
@@ -197,10 +197,14 @@ - (void)listOnSpotlightWithCallback:(callbackWithUrl)callback {
         publiclyIndexable = YES;
     }
     
+    NSMutableDictionary *metadataAndProperties = [self.metadata mutableCopy];
+    metadataAndProperties[BRANCH_LINK_DATA_KEY_CANONICAL_IDENTIFIER] = self.canonicalIdentifier;
```
Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/BranchUniversalObject.m]: check for existence, can't put in nil properties

**Comment by Sarkar** [comment]
i dunno how strongly we feel about `{ }` on `if` statements - otherwise looks reasonable

**Comment by ahmednawar** [comment]
@Sarkar the codebase has `{ }` on `if` statements even for one liners..will change to be consistent

**Comment by Sarkar** [comment]
:+1:

**Pull request merged**